### PR TITLE
Kth largest/smallest element program in C++

### PR DESCRIPTION
**Pull request by 1711shashank**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes I have read the Contributing Guidelines

### Description

Kth largest element in an array
Kth smallest element in an array

### Checklist

- [✔ ] I've read the contribution guidelines.
- [ ✔] I've checked the issue list before deciding what to submit.
- [✔ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

https://github.com/TesseractCoding/NeoAlgo/issues/1425

### CODE REVIEW

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
+//Q Kth largest element
```
Comment (HarshCasper) [inline C-Plus-Plus/other/Kth largest element.cpp]: Have a beginner description about the Algorithms and the Code Implementation in a Multi-Line Comment.

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
+//Q Kth largest element 
+
+// Given an array a[] and a number K where K is smaller than size of array, 
+// the task is to find the Kth largest element in the given array.
+// It is given that all array elements are distinct.
+
+//Input
```
Comment (HarshCasper) [inline C-Plus-Plus/other/Kth largest element.cpp]: Mention I/O Format at the end of the Code

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    cin >> n >> k;
+    int a[n];
+
+    for (int i = 0; i < n; i++){
+        cin >> a[i];
+    }
+
+    // a={7, 5, 9, 8, 6, 4, 10}
+        
+    priority_queue<int, vector<int>, greater<int>> minh;
```
Comment (HarshCasper) [inline C-Plus-Plus/other/Kth largest element.cpp]: Have all your primary logic in another function. The main() function should strictly serve as your Driver Code.

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+// Output
+// 8
+
+#include <bits/stdc++.h>
+#include <iostream>
+using namespace std;
+
+int main()
+{
+    int n, k;
```
Comment (HarshCasper) [inline C-Plus-Plus/other/Kth largest element.cpp]: Use proper Variable Names. Prefer using Snake Case or Camel Case convention.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by 1711shashank** [comment]
@HarshCasper 
Resolved Changes

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    kthLargestElement(a,size,k);  
+    
+    return 0;
+}
+
+// Input 
+// size=7 k=3
+// 7 5 9 8 6 4 10
+// Output
+// 8
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/other/Kth largest element.cpp]: Add time and space complexity

File: C-Plus-Plus/other/Kth smallest element.cpp [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    kthSmallestElement(arr, size, k);  
+    
+    return 0;
+}
+
+// Input 
+// size=7 k=3
+// 7 5 9 8 6 4 10
+// Output
+// 6
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/other/Kth smallest element.cpp]: Add time and space complexity

File: C-Plus-Plus/other/Sort a nearly sorted array.cpp [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+/*
+Input 
+6 3
+2 6 3 12 56 8
+
+Output
+12 56 8 6 3 2
+*/
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/other/Sort a nearly sorted array.cpp]: Add time and space complexity

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 8]
```diff
@@ -0,0 +1,49 @@
+/*
+Q Kth largest element 
+
+Given an array a[] and a number K where K is smaller than size of array, 
+the task is to find the Kth largest element in the given array.
+It is given that all array elements are distinct. */
+
+
+#include <bits/stdc++.h>
+#include <iostream>
```
Comment (atarax665) [inline C-Plus-Plus/other/Kth largest element.cpp]: You don't need to add ```#include <iostream>``` if you have already added ```#include <bits/stdc++.h>```. Make the change in both the programs.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Input and output should be accompanied with proper messages.
i.e -
```
cout<<"Enter the size of the array : "<<endl;
cin>>size;
```

**Comment by 1711shashank** [comment]
Resolved Changes

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+    priority_queue <int, vector<int>, greater<int>> minHeap;
+    
+    for(int i = 0; i < size; i++){
+        minHeap.push(a[i]);
+        
+        if(minHeap.size() > k){
+            minHeap.pop();
+        }
+    }
+    cout<< minHeap.top();
```
Comment (atarax665) [inline C-Plus-Plus/other/Kth largest element.cpp]:
```suggestion
    cout<< "Kth largest element in the array is: "<< minHeap.top();
```

File: C-Plus-Plus/other/Kth smallest element.cpp [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+    priority_queue <int> maxHeap;
+    
+    for(int i = 0; i < size; i++){
+        maxHeap.push(arr[i]);
+        
+        if(maxHeap.size() > k){
+            maxHeap.pop();
+        }
+    }
+    cout<< maxHeap.top();
```
Comment (atarax665) [inline C-Plus-Plus/other/Kth smallest element.cpp]:
```suggestion
    cout<< "Kth smallest element in the array is: "<< maxHeap.top();
```

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+    cout << "Enter the value of k : " << endl;
+    cin>> k;
+    
+    int a[size];
+
+    for (int i = 0; i < size; i++){
+        cin >> a[i];
+    }
+
+    // a={7, 5, 9, 8, 6, 4, 10}
```
Comment (atarax665) [inline C-Plus-Plus/other/Kth largest element.cpp]: Remove this comment as you have already added Sample I/O at the end.

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+    int size, k; 
+    
+    cout << "Enter the size of an Array : " << endl;
+    cin >> size;
+
+    cout << "Enter the value of k : " << endl;
+    cin>> k;
+    
+    int a[size];
+
```
Comment (atarax665) [inline C-Plus-Plus/other/Kth largest element.cpp]:
```suggestion
    cout<< "Enter the elements of the array: "<<endl;
```

File: C-Plus-Plus/other/Kth largest element.cpp [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+    // a={7, 5, 9, 8, 6, 4, 10}
+
+    kthLargestElement(a,size,k);  
+    
+    return 0;
+}
+
+/*
+Input 
+size=7 k=3
```
Comment (atarax665) [inline C-Plus-Plus/other/Kth largest element.cpp]: Format Sample I/O according to the recent changes that you have made in taking input and displaying output. Sample I/O should be exactly same as what would be displayed once the program is run in some IDE. Make changes in both the program.

File: C-Plus-Plus/other/Sort a nearly sorted array.cpp [hunk 14]
```diff
@@ -0,0 +1,61 @@
+/* Sort a nearly sorted (or K sorted) array
```
Comment (atarax665) [inline C-Plus-Plus/other/Sort a nearly sorted array.cpp]: Raise a separate PR for this program and do raise an issue for the same if its doesn't exist currently.

File: C-Plus-Plus/other/Kth smallest element.cpp [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+    cout << "Enter the value of k : " << endl;
+    cin>> k;
+
+    int arr[size];
+
+    for (int i = 0; i < size; i++){
+        cin >> arr[i];
+    }
+
+    // arr={7, 5, 9, 8, 6, 4, 10}
```
Comment (atarax665) [inline C-Plus-Plus/other/Kth smallest element.cpp]: Remove this comment as you have already added Sample I/O at the end.

File: C-Plus-Plus/other/Kth smallest element.cpp [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+    int size, k;
+     
+    cout << "Enter the size of an Array : " << endl;
+    cin >> size;
+
+    cout << "Enter the value of k : " << endl;
+    cin>> k;
+
+    int arr[size];
+
```
Comment (atarax665) [inline C-Plus-Plus/other/Kth smallest element.cpp]:
```suggestion
    cout<< "Enter the elements of the array: "<<endl;
```

**Review by atarax665 — CHANGES_REQUESTED** [review]
Change the file name to ```Kth_largest_element.cpp``` and ```Kth_smallest_element.cpp``` and make the changes in ```README.md``` respectively.

**Review by vatsalkesarwani12 — APPROVED** [review]
👍🏻

**Review by atarax665 — APPROVED** [review]
:+1: Good Work

**Pull request merged**